### PR TITLE
feat: Add onTapAlwaysCalled in TextFormField

### PR DESCRIPTION
**Pull request by lsaudon**
Add onTapAlwaysCalled in TextFormField

*List which issues are fixed by this PR. You must list at least one issue. An issue is not required if the PR fixes something trivial like a typo.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test/material/text_form_field_test.dart [hunk 1]
```diff
@@ -1512,6 +1512,27 @@ void main() {
...
   });
 
+  testWidgetsWithLeakTracking('Passes onTapAlwaysCalled to onTapAlwaysCalled TextField', (WidgetTester tester) async {
+    const bool onTapAlwaysCalled = true;
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Material(
+          child: Center(
+            child: TextFormField(
+              onTapAlwaysCalled: onTapAlwaysCalled,
```
Comment (Renzo-Olivares) [inline packages/flutter/test/material/text_form_field_test.dart:1522]: nit: I think we can just set this to `true` without the local variable.

File: packages/flutter/test/material/text_form_field_test.dart [hunk 2]
```diff
@@ -1512,6 +1512,27 @@ void main() {
...
+          ),
+        ),
+      ),
+    );
+
+    final Finder textFieldFinder = find.byType(TextField);
+    expect(textFieldFinder, findsOneWidget);
+
+    final TextField textFieldWidget = tester.widget(textFieldFinder);
+    expect(textFieldWidget.onTapAlwaysCalled, onTapAlwaysCalled);
```
Comment (Renzo-Olivares) [inline packages/flutter/test/material/text_form_field_test.dart:1533]: should be: `expect(textFieldWidget.onTapAlwaysCalled, isTrue);`

**Review by Renzo-Olivares — APPROVED** [review]
LGTM

**Review by cbracken — APPROVED** [review]
LGTM stamp from a Japanese personal seal

**Pull request merged**